pkg/metrics/instance/modal_manager.go

```diff
@@ -68,6 +69,30 @@ type ModalManager struct {
 	wrapped, active Manager
 }
 
+// ApplyConfigs is used to batch configurations for performance instead of the singular ApplyConfig
+func (m *ModalManager) ApplyConfigs(configs []Config) error {
+	m.mut.Lock()
+	defer m.mut.Unlock()
+
+	err := m.active.ApplyConfigs(configs)
+	var bae BatchApplyError
+	if err != nil {
+		errors.As(err, &bae)
+	}
```

Comment:
nit: `_ = errors.As(err, &bae)`  to make it clear we're just trying to blindly convert.